scene/2d/camera_2d.cpp

```diff
@@ -593,15 +596,26 @@ float Camera2D::get_h_offset() const {
 
 void Camera2D::_set_old_smoothing(float p_enable) {
 	//compatibility
-	if (p_enable > 0) {
-		smoothing_enabled = true;
-		set_follow_smoothing(p_enable);
-	}
+	set_enable_follow_smoothing(p_enable > 0);
+	set_follow_smoothing(p_enable);
 }
 
 void Camera2D::set_enable_follow_smoothing(bool p_enabled) {
 
+	// watch for situation where an pre-enabled camera is added to the tree
+	// processing must be resumed and bypass this noop check
```

Comment:
There's a call to `_update_process_mode()` in `NOTIFICATION_ENTER_TREE` so I guess it's fine?
What could be tricky is if `smoothing_enabled` and `smoothing_active` defaulted to `true`, as you could end up in a situation where `smoothing_active` is not properly initialized to match editor status if `set_enable_follow_smoothing()` is never called.

But here it should be fine, `false` by default is safe, and turning smoothing on should have to go through `set_enable_follow_smoothing()`.

As a side note for `master`, we should likely rename this to `set_follow_smoothing_enabled()`.

Reply:
I've added to the note that it is working fine now but to watch for future bugs due to this. We could remove the comment but if so, I reserve the right to say 'I told you so' in 2025. :grinning: